File: cp_rust/src/bin/abc287-b-postal-card.rs
// Created by Ayush Biswas at 2026/07/07 11:35
// https://atcoder.jp/contests/adt_easy_20260625_1/tasks/abc287_b
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;
use std::collections::HashSet;

sol! {
    fn solution(
        [n, m]: [usize; 2],
        s: [usize]; n,
        t: [usize]; m,
    ) -> usize {
        let ts: HashSet<usize> = HashSet::from_iter(t.into_iter());
        s.into_iter().count_where(|si| {
            ts.contains(&(si % 1000))
        })
    }
}
// @code end

File: cp_nim/src/bin/abc287-b-postal-card.nim
# Created by Ayush Biswas at 2026/07/18 18:02
# https://atcoder.jp/contests/adt_easy_20260625_1/tasks/abc287_b
import ../lib/cpio

# @code begin

import sets

var r = newReader()

let
  (n, m) = r[int, int]
  s = r[seq[string], n]
  t = r[seq[string], m].toHashSet

var res: int
for w in s:
  if t.contains(w[^3 ..^ 1]):
    inc res
echo res

# @code end
